scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,119 @@
+const fs = require('fs');
+const yaml = require('js-yaml');
+const { Parser } = require('@asyncapi/parser');
+const mergePatch = require('json-merge-patch');
+const jsonpointer = require('jsonpointer');
+const parser = new Parser();
+
+// Read the markdown file
+const markdownContent = fs.readFileSync('../../spec/asyncapi.md', 'utf8');
+
+// Function to extract comments and examples from the markdown content
+function extractCommentsAndExamples(content) {
+  const combinedRegex = /<!--\s*asyncapi-example-tester:\s*({.*?})\s*-->\s*\n```(.*)?\n([\s\S]*?)\n```/g;
+  let match;
+  const combinedData = [];
+
+  while ((match = combinedRegex.exec(content)) !== null) {
+    try {
+      const json = JSON.parse(match[1]);
+      const format = match[2].trim();
+      const exampleContent = match[3].trim();
+
+      let example;
+      if (format === 'json') {
+        example = JSON.parse(exampleContent);
+      } else if (format === 'yaml') {
+        example = yaml.load(exampleContent);
+      } else {
+        throw new Error(`Unsupported format: ${format}`);
+      }
+
+      combinedData.push({
+        name: json.name,
+        json_pointer: json.json_pointer,
+        example: example,
+        format: format,
+      });
+    } catch (e) {
+      console.error("Failed to parse comment JSON or example:", match[1], e);
+      process.exit(1);
+    }
+  }
+
+  return combinedData;
+}
+
+// Extract comments and examples from the markdown file
+const combinedData = extractCommentsAndExamples(markdownContent);
+
+// Function to apply JSON Merge Patch updates to the document
+function applyUpdates(updates, baseDoc) {
+  updates.forEach(update => {
+    try {
+      const jsonPointerPath = update.json_pointer;
```

Comment:
Nitpick: remove this line. IMHO it is better to know the example comes from the object in the comment example

```suggestion
```